pages/mapping.js

```diff
@@ -47,16 +49,17 @@ export default function mapping() {
 		}
 
 		// Initialize cookie if not present
-		if(!cookies.uuid){
+		if(!session && !cookies.uuid){
 			const userId = uuidv4();
 			setCookie('uuid', userId, { path: '/', maxAge: 2592000 }); // maxAge: 30 days
 			addUser(userId);
 		}
 
 		// Initialize the set of questions
 		async function fetchData() {
+			const user_id = session ? session.user.id : cookies.uuid
```

Comment:
We have to be careful here. If session is not set, and because setCookie is asynchronous (from L54 above), it is possible that when the program executes this line cookies.uuid is not set yet.

We have observed similar unexpected behavior in the past.

Reply:
Yes, I’ve noticed it as well. `fetchData` is executed before the uuid is set, so a bug appears when the user starts to map without signing in. But this does not relate to the changes I’ve made.

Reply:
Fixed in d3b4d6a